### PR TITLE
 vstart.sh: add an option to use crimson-osd

### PR DESCRIPTION
**Pull request by tchaikov**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/vstart.sh [hunk 1]
```diff
@@ -754,7 +765,7 @@ EOF
 	    echo "{\"cephx_secret\": \"$OSD_SECRET\"}" > $CEPH_DEV_DIR/osd$osd/new.json
             ceph_adm osd new $uuid -i $CEPH_DEV_DIR/osd$osd/new.json
 	    rm $CEPH_DEV_DIR/osd$osd/new.json
-            $SUDO $CEPH_BIN/ceph-osd -i $osd $ARGS --mkfs --key $OSD_SECRET --osd-uuid $uuid
+            $SUDO $CEPH_BIN/$ceph_osd $crimson_args -i $osd $ARGS --mkfs --key $OSD_SECRET --osd-uuid $uuid
```
Comment (athanatos) [inline src/vstart.sh]: s/crimson_args/extra_osd_args?

Seems like it could be used to pass anything to the osd.

**Comment by athanatos** [comment]
LGTM (I don't feel strongly about my comment)

File: src/vstart.sh (hunk 1 above)
Comment (tchaikov) [reply src/vstart.sh]: @athanatos fixed and repushed.

**Comment by liu-chunmei** [comment]
It is OK to me,  how to use --osd-args?  --osd-args --smp 1? or others?

**Comment by tchaikov** [comment]
@liu-chunmei i think it should be:

```
vstart.sh .... --osd-args "--smp 1"
```

File: src/vstart.sh [hunk 2]
```diff
@@ -227,6 +230,13 @@ case $1 in
     --short )
 	    short=1
 	    ;;
+    --crimson )
+        ceph_osd=crimson-osd
+        ;;
+    --osd-args )
```
Comment (liu-chunmei) [inline src/vstart.sh]: do we need in some places to set extra_osd_args to be empty? otherwise, if no --osd-args input, what happens?

Comment (tchaikov) [reply src/vstart.sh]: it's empty. please try this in your testbed.
```sh
echo ${good_news_everyone_today_is_Saturday}
```

**Pull request merged**